servers/services/src/main/java/org/projectnessie/services/authz/AccessContext.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.services.authz;
+
+import java.security.Principal;
+
+/** Provides some context about a role/principal that accesses Nessie resources. */
+public interface AccessContext {
+  /** Provide a unique id for the operation being validated (for correlation purposes). */
+  String operationId();
+
+  /** Provide the user identity. */
+  Principal user();
```

Comment:
perhaps instead of adding contents.id to the interface it could be added here as Optional<String>?

Reply:
that means that non-related stuff "leaks" into this, which I'm not really happy with